src/app/DrawingEngine.ts

```diff
@@ -19,6 +19,24 @@ export class DrawingEngine extends BaseDrawingEngine<AvailablePrograms> {
     }
   }
 
+  public updateDrawing(currentSegment: number[]) {
+    if (currentSegment.length > 0) {
+      this.drawLineSegment(currentSegment)
+    }
+  }
+
+  private drawLineSegment(segment: number[]) {
+    const gl = this.gl
+    const buffer = gl.createBuffer()
+    gl.bindBuffer(gl.ARRAY_BUFFER, buffer)
+    gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(segment), gl.DYNAMIC_DRAW)
+
+    const position = gl.getAttribLocation(this.programs.lineDrawing.program, "position")
+    gl.enableVertexAttribArray(position)
+    gl.vertexAttribPointer(position, 2, gl.FLOAT, false, 0, 0)
+    gl.drawArrays(gl.LINE_STRIP, 0, segment.length / 2)
+  }
```

Comment:
We can just use `drawLine` instead of this and it should work the same. I double checked and it's the same lines of code being called, the only difference being drawLine makes sure that the color is set from the color context, and this uses `gl.DYNAMIC_DRAW` while the other uses `gl.STREAM_DRAW`. We can add a parameter to change the draw type but it doesn't seem like it's probably going to change very much.

Reply:
I committed a change for this